src/hooks/useStatsig.tsx

```diff
@@ -1,21 +1,36 @@
-import { StatsigClient } from '@statsig/js-client';
-import { StatsigProvider, useStatsigClient } from '@statsig/react-bindings';
+import { useEffect, useState } from 'react';
 
-export enum StatSigFlags {
-  // When adding a flag here, make sure to add an analytics tracker in useAnalytics.ts
-  ffSkipMigration = 'ff_skip_migration',
-}
+import { StatSigFlags, StatsigConfigType } from '@/types/statsig';
+import { StatsigClient } from '@statsig/js-client';
+import {
+  StatsigProvider as StatsigProviderInternal,
+  useStatsigClient,
+} from '@statsig/react-bindings';
 
-export const StatSigProvider = ({ children }: { children: React.ReactNode }) => {
-  const client = new StatsigClient(`${import.meta.env.VITE_STATSIG_CLIENT_KEY}`, {
-    // TODO: fill in with ip address
-  });
-  client.initializeSync();
+import { statsigClientPromise } from '@/lib/statsig';
 
-  return <StatsigProvider client={client}> {children} </StatsigProvider>;
+export const StatsigProvider = ({ children }: { children: React.ReactNode }) => {
+  const [client, setClient] = useState<StatsigClient | null>(null);
+  useEffect(() => {
+    const setAsyncClient = async () => {
+      const statsigClient = await statsigClientPromise;
+      setClient(statsigClient);
+    };
+    setAsyncClient();
+  }, [statsigClientPromise]);
+  // if no client, render without provider until a client exists
+  if (!client) return <div>{children}</div>;
```

Comment:
what if the promise fails / throws? we never render the app?

Reply:
if we don't get a client, we pass the children through the component without the provider so it should just behave as if there's no statsig provider, right?

or are you implying we need a `try/catch` in our useEffect hook?

Reply:
oh yeah I totally misread the code

Reply:
should we be logging if the client fails/throws?